unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,235 @@
+        with pytest.raises(RedisAPIException,
+                           match="There is no matching command response"
+                                 " in _blocking_command_responses."):
+            redis_client_auto_register._check_buffer_message_matching_command_and_id(data)
+
+    @staticmethod
+    def test_register(redis_client_auto_register):
+        """Check if the redis.db.publish is called with correct arguments."""
+        redis_client_auto_register.redis_db.publish = MagicMock()
+        redis_client_auto_register.register(is_blocking=False)
+        transaction_id = redis_client_auto_register._blocking_command_responses[
+            "register"]["transaction_id"]
+        data = {"name": AREA_ID, "transaction_id": transaction_id}
+        redis_client_auto_register.redis_db.publish.assert_called_once_with(
+            f"{AREA_ID}/register_participant", json.dumps(data))
+
+    @staticmethod
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_register_self_active_false_throws_exception(mock_wait_until_timeout_blocking,
+                                                         redis_client_auto_register):
+        """Check whether if is active is false throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API registration process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background "
+                                 "and will notify you as soon as the "
+                                 "registration has been completed."):
+            redis_client_auto_register.register(is_blocking=True)
+
+    @staticmethod
+    def test_register_self_active_true_throw_exception(redis_client_auto_register):
+        """Check whether if is active is true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API is already registered to the market."):
+            redis_client_auto_register.is_active = True
+            redis_client_auto_register.register(is_blocking=False)
+
+    @staticmethod
+    def test_unregister(redis_client_auto_register):
+        """Check whether the redis.db.publish is called with correct arguments."""
+        redis_client_auto_register.redis_db.publish = MagicMock()
+        redis_client_auto_register.is_active = True
+        redis_client_auto_register.unregister(is_blocking=False)
+        transaction_id = redis_client_auto_register._blocking_command_responses[
+            "unregister"]["transaction_id"]
+        data = {"name": AREA_ID, "transaction_id": transaction_id}
+        redis_client_auto_register.redis_db.publish.assert_called_once_with(
+            f"{AREA_ID}/unregister_participant",
+            json.dumps(data))
+
+    @staticmethod
+    def test_unregister_is_active_false_throws_exception(redis_client_auto_register):
```

Comment:
Nit: it would be better to refactor test names to have more consistency:
- `test_register_self_active_false_throws_exception`
- `test_register_self_active_true_throw_exception`
- `test_unregister_is_active_false_throws_exception`
- `test_unregister_is_active_true_throws_exception`

As you see, the naming is different. This is a small nit anyway.

Reply:
Sure, Thank you